diff --git a/src/messageSocket.native.ts b/src/messageSocket.native.ts
index 1d7c93325d1af908040d6f5b7ccf7e990b040c3c..9ce5975b9f483028322dc8fb4e88b535418021e2 100644
--- a/src/messageSocket.native.ts
+++ b/src/messageSocket.native.ts	
@@ -1,7 +1,8 @@
 /* eslint-env browser */
 
 function createWebSocketConnection(path: string = '/message') {
-  const getDevServer = require('react-native/Libraries/Core/Devtools/getDevServer');
+  const _getDevServerModule = require('react-native/Libraries/Core/Devtools/getDevServer');
+  const getDevServer = _getDevServerModule.default ?? _getDevServerModule;
   const devServer = getDevServer();
   if (!devServer.bundleLoadedFromServer) {
     throw new Error('Cannot create devtools websocket connections in embedded environments.');
@@ -9,7 +10,8 @@ function createWebSocketConnection(path: string = '/message') {
 
   const devServerUrl = new URL(devServer.url);
   const serverScheme = devServerUrl.protocol === 'https:' ? 'wss' : 'ws';
-  const WebSocket = require('react-native/Libraries/WebSocket/WebSocket');
+  const _WebSocketModule = require('react-native/Libraries/WebSocket/WebSocket');
+  const WebSocket = _WebSocketModule.default ?? _WebSocketModule;
   return new WebSocket(`${serverScheme}://${devServerUrl.host}${path}`);
 }
 
